core/src/main/java/org/apache/iceberg/rest/RESTCatalogProperties.java

```diff
@@ -35,6 +35,14 @@ private RESTCatalogProperties() {}
 
   public static final String PAGE_SIZE = "rest-page-size";
 
+  // Enable a lightweight client-side reconciliation flow for REST commits that add a new snapshot
+  // when the server responds with CommitStateUnknown (e.g., transient 5xx). When enabled, the
+  // client will refresh the table and verify the expected snapshot is present in history; if so,
+  // the commit is treated as successful without re-executing.
+  public static final String RECONCILE_ON_COMMIT_STATE_UNKNOWN =
```

Comment:
Just checking, do we actually need a config for this? It seems like it should be on all the time.

Reply:
If there are no objections, I’ll remove the property and make reconciliation always-on.

Reply:
+1 on removing the config